Trakt/Api/TraktApi.cs

```diff
@@ -819,8 +849,38 @@ private async Task<TraktSyncResponse> SendEpisodePlaystateUpdatesInternalAsync(I
 
         using (var response = await PostToTrakt(url, data, traktUser, cancellationToken).ConfigureAwait(false))
         {
-            return await JsonSerializer.DeserializeAsync<TraktSyncResponse>(response, _jsonOptions, cancellationToken).ConfigureAwait(false);
+            var tsr = await JsonSerializer.DeserializeAsync<TraktSyncResponse>(response, _jsonOptions, cancellationToken).ConfigureAwait(false);
+            if (useProviderIDs && tsr.NotFound.Episodes.Count > 0)
+            {
+                // send subset of episodes back to trakt to try without ids
+                _logger.LogDebug("Resend episodes playstate update, without episode IDs");
+                await SendEpisodePlaystateUpdatesInternalAsync(FindNotFoundEpisodes(episodeChunk, tsr), traktUser, seen, cancellationToken, false).ConfigureAwait(false);
+            }
+
+            return tsr;
+        }
+    }
+
+    private List<Episode> FindNotFoundEpisodes(IEnumerable<Episode> episodeChunk, TraktSyncResponse traktSyncResponse)
+    {
+        // episodes not found. if using IDs, try again without them
+        List<Episode> episodes = new List<Episode>();
+        // build a list of unfound episodes with ids
+        foreach (TraktEpisode traktEpisode in traktSyncResponse.NotFound.Episodes.Where(i => HasAnyProviderTvIds(i.Ids)))
+        {
+            // find matching episode in JF based on ids provide
+            var notFoundEpisode = episodeChunk.First(e => e.GetProviderId(MetadataProvider.Imdb) == traktEpisode.Ids.Imdb
+                && e.GetProviderId(MetadataProvider.Tmdb) == traktEpisode.Ids.Tmdb?.ToString(CultureInfo.InvariantCulture)
+                && e.GetProviderId(MetadataProvider.Tvdb) == traktEpisode.Ids.Tvdb?.ToString(CultureInfo.InvariantCulture)
+                && e.GetProviderId(MetadataProvider.TvRage) == traktEpisode.Ids.Tvrage?.ToString(CultureInfo.InvariantCulture));
```

Comment:
Shouldn't these all be `||`? This would only return the episode if all provider ids match

Reply:
In my testing Trakt was returning all the IDs I would send, so this would match. But I'm not opposed to switching it to `||`